src/components/scenes/CreateWalletImportScene.tsx

```diff
@@ -1,167 +1,172 @@
-            buttons={{ ok: { label: s.strings.string_ok } }}
-            message={error.message}
-            title={s.strings.create_wallet_failed_import_header}
-          />
-        ))
-      )
-      .then(() => this.setState({ isProcessing: false }))
-  }
+  const account = useSelector(state => state.core.account)
+  const { currencyConfig } = account
 
-  onChangeText = (input: string) => {
-    this.setState({ input })
-  }
+  const [importText, setImportText] = React.useState('')
+
+  const handleNext = useHandler(async () => {
+    const cleanImportText = importText.trim()
+
+    // Test imports
+    const { newWalletItems } = splitCreateWalletItems(createWalletList)
 
-  render() {
-    const { route } = this.props
-    const { error, isProcessing, input } = this.state
-    const { selectedWalletType } = route.params
-    const { walletType } = selectedWalletType
-
-    const specialCurrencyInfo = getSpecialCurrencyInfo(walletType)
-    if (!specialCurrencyInfo.isImportKeySupported) throw new Error()
-    const instructionSyntax = specialCurrencyInfo.isImportKeySupported.privateKeyInstructions
-    const labelKeySyntax = specialCurrencyInfo.isImportKeySupported.privateKeyLabel
-    return (
-      <SafeAreaView>
-        <View style={styles.scene}>
-          <Gradient style={styles.gradient} />
-          <View style={styles.view}>
-            <View style={styles.createWalletPromptArea}>
-              <Text style={styles.instructionalText}>{instructionSyntax}</Text>
-            </View>
-            <FormField
-              autoFocus
-              autoCorrect={false}
-              onChangeText={this.onChangeText}
-              label={labelKeySyntax}
-              value={input}
-              returnKeyType="next"
-              onSubmitEditing={this.handleNext}
-              multiline
-              error={error}
-            />
-            <View style={styles.buttons}>
-              <PrimaryButton style={styles.next} onPress={this.handleNext} disabled={isProcessing}>
-                {isProcessing ? (
-                  <ActivityIndicator color={THEME.COLORS.ACCENT_MINT} />
-                ) : (
-                  <PrimaryButton.Text>{s.strings.string_next_capitalized}</PrimaryButton.Text>
-                )}
-              </PrimaryButton>
-            </View>
-          </View>
-        </View>
-      </SafeAreaView>
+    const pluginIds = newWalletItems.map(item => item.pluginId)
+
+    // Loop over plugin importPrivateKey
+    const promises = pluginIds.map(
+      async pluginId =>
+        await currencyConfig[pluginId].importKey(cleanImportText).catch(e => {
+          console.warn('importKey failed', e)
+        })
     )
-  }
+
+    const results = await Promise.all(promises)
+
+    const successMap: { [pluginId: string]: JsonObject } = {}
+
+    for (const [i, keys] of results.entries()) {
+      if (typeof keys === 'object') {
+        // Success
+        successMap[pluginIds[i]] = keys
+      }
+    }
```

Comment:
Oooh, I have learned something new. I have always done the oldschool `for  (let i = 0; i < array.length; ++i)` when I needed to iterate over an array and needed the loop counter too. I did not know you could use `array.entries()` to get access to the loop counter in a `for of` loop. Neato.